module/modmanager/manager.go

```diff
@@ -202,37 +262,40 @@ func (mgr *Manager) startModule(ctx context.Context, mod *module) error {
 	if mod.dataDir != "" {
 		mgr.logger.Infof("Creating data directory %q for module %q", mod.dataDir, mod.cfg.Name)
 		if err := os.MkdirAll(mod.dataDir, 0o750); err != nil {
-			return errors.WithMessage(err, "error while creating data directory for module "+mod.cfg.Name)
+			return errlib.WithMessage(err, "error while creating data directory for module "+mod.cfg.Name)
 		}
 	}
 
 	if err := mod.startProcess(mgr.restartCtx, mgr.parentAddr,
 		mgr.newOnUnexpectedExitHandler(mod), mgr.logger, mgr.viamHomeDir); err != nil {
-		return errors.WithMessage(err, "error while starting module "+mod.cfg.Name)
+		return errlib.WithMessage(err, "error while starting module "+mod.cfg.Name)
 	}
 
 	if err := mod.dial(); err != nil {
-		return errors.WithMessage(err, "error while dialing module "+mod.cfg.Name)
+		return errlib.WithMessage(err, "error while dialing module "+mod.cfg.Name)
 	}
 
 	if err := mod.checkReady(ctx, mgr.parentAddr, mgr.logger); err != nil {
-		return errors.WithMessage(err, "error while waiting for module to be ready "+mod.cfg.Name)
+		return errlib.WithMessage(err, "error while waiting for module to be ready "+mod.cfg.Name)
 	}
 
-	mod.registerResources(mgr, mgr.logger)
-	mgr.modules[mod.cfg.Name] = mod
```

Comment:
this "registration" part is annoying to make thread-safe. however, it's fast compared to the previous steps, so I decided to pull it out and run in serial instead of trying to parallelize it.